Userland/Utilities/pls.cpp

```diff
@@ -0,0 +1,196 @@
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <AK/String.h>
+#include <AK/StringBuilder.h>
+#include <AK/Types.h>
+#include <AK/Vector.h>
+#include <LibCore/Account.h>
+#include <LibCore/ArgsParser.h>
+#include <LibCore/File.h>
+#include <LibCore/GetPassword.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+
+static constexpr int line_buff_length = 256;
+
+// Function Definitions
+extern "C" int main(int arch, char** argv);
+int unveil_paths(const char*);
+
+// Unveil paths, given that the current
+int unveil_paths(const char* command)
+{
+    // Get the system path, split it and attempt to unveil all the paths.
+    // We do NOT error out on an invalid path
+    auto paths = String(getenv("PATH")).split(':');
+    int num_unveils = 0;
+    char path_buf[256];
+
+    // Unveil each path
+    for (const auto path : paths) {
+        if (unveil(path.characters(), "x") == 0)
+            num_unveils++;
+    }
+
+    // Now unveil the command
+    auto command_path = realpath(command, &path_buf[0]);
+    if (command_path) {
+        if (unveil(command_path, "x") == 0)
+            num_unveils++;
+    }
+
+    return num_unveils;
+}
+
+// <kling> linusg: quaker: "please" feels quite long, how about "pls" :P
+// <kling> "pls rm -r crap" has a nice ring to it
+// lol
+int main(int argc, char** argv)
+{
+    Vector<const char*> command;
+    Core::ArgsParser args_parser;
+    args_parser.add_positional_argument(command, "Name of command to run at elevated privilege level", "cmd");
+
+    args_parser.parse(argc, argv);
+
+    if (pledge("stdio tty rpath exec id", nullptr) < 0) {
+        perror("pledge");
+        return 1;
+    }
+
+    if (seteuid(0) < 0) {
+        perror("seteuid");
+        return 1;
+    }
+
+    if (unveil("/etc", "r") < 0) {
+        perror("unveil");
+        return 1;
+    }
+
+    const char* username = getlogin();
+    Vector<String> sudoers;
+    auto sudoers_fd = fopen("/etc/sudoers", "r");
+    if (!sudoers_fd) {
+        warnln("couldn't open /etc/sudoers!");
+        return 1;
+    }
+
+    // Okay, so we have a handle to /etc/sudoers, now let's read it in line by line to get a
+    // list of all of the users contained within
+    char buffer[line_buff_length];
+    while (fgets(buffer, line_buff_length, sudoers_fd)) {
+        // Ignore comments
+        if (buffer[0] == '#')
+            continue;
+
+        buffer[strcspn(buffer, "\n")] = '\0';
+        sudoers.append(String(buffer));
+    }
+
+    if (!sudoers.contains_slow(username)) {
+        warnln("{} is not in the sudoers file. This incident will be reported :^)", username);
```

Comment:
Where will this be reported? Or is it left in for the memes? :P

Reply:
Perhaps eventually? We have `syslog()` but it is currently unused.

Reply:
Please abstain from memeing in user-facing parts of the system.

Reply:
Sorry I wasn't aware this was memeing as it gives the same output as sudo sans :^)